src/firefly/js/ui/FileUpload.jsx

```diff
@@ -207,10 +207,12 @@ export function doUpload(file, params={}) {
     return fetchUrl(UL_URL, options).then( (response) => {
         return response.text().then( (text) => {
             // text is in format ${status}::${message}::${message}::${cacheKey}::${analysisResult}
-            const result = text.split('::');
-            const [status, message, cacheKey, fileFormat] = result.slice(0, 4);
-            const analysisResult = result.slice(4).join('::');
-            return {status, message, cacheKey, fileFormat, analysisResult};
+            let [status, message, cacheKey, analysisResult, ...rest] = text.split('::');
+            if (rest.length > 0) {
```

Comment:
`::` is used as a separator token for status, message, etc.  But in some cases, `analysisResult` may also contain the character `::`.  Usually it's in one of the comments.
JavaScript split(s, limit) is different from Java where it would discard the remaining string when limit is reached.  Therefore, I had to split the whole string, and then put back what was incorrectly split.